### ISSUE TITLE
gdscript‘s ”enum“ have some problems

### ISSUE THREAD

**Issue by vmjcv** [godotengine/godot#36672]
**Godot version:**
v4.0.dev.custom_build.a596d7615
**OS/device including version:**
windows 10 1903
**Issue description:**
use enum in my project, the error is "Invalid get index 'One'(on base:'Dictionary')"
**Steps to reproduce:**
try this gdscript in project

```
enum warehouse_key {
	One, 
	Two
}

func _init():
	print(warehouse_key.One)
```

**Minimal reproduction project:**

**Comment by cat-kitten** [comment]
I can't get it to work in any way (even if I follow the [official docs](https://docs.godotengine.org/en/3.1/getting_started/scripting/gdscript/gdscript_basics.html) for enums).

btw here is minimal reproduction project:
[asd.zip](https://github.com/godotengine/godot/files/4270283/asd.zip)

**Comment by Calinou** [comment]
@cat-kitten Which Godot version are you using?

**Comment by cat-kitten** [comment]
My own fork of the master branch ([this](https://github.com/cat-kitten/godot/commit/8ffa35e44f93321d618b81ec48817f5de8668a83))

**Comment by Calinou** [comment]
Maybe enums were broken as part of the GDScript refactoring that's currently going on.

**Comment by vmjcv** [comment]
ok，If this problem is solved, can you inform us

**Comment by akien-mga** [comment]
@ThakeeNathees You closed your PR, is this fixed by the "GDScript 2.0" PR?

**Comment by ThakeeNathees** [comment]
@akien-mga yeah, enums have completely refactored and It has fixed. (also named enums can be used as a type now) this issue may closed.
